runtime/src/block_min_prioritization_fee_cache_update.rs

```diff
@@ -0,0 +1,14 @@
+use solana_sdk::{clock::Slot, transaction::SanitizedTransaction};
+
+pub trait BlockMinPrioritizationFeeCacheUpdate {
+    /// Update block's min prioritization fee with `txs`,
+    /// Returns updated min prioritization fee for `slot`
+    fn update_transactions<'a>(
+        &mut self,
+        slot: Slot,
+        txs: impl Iterator<Item = &'a SanitizedTransaction>,
+    ) -> Option<u64>;
+
+    /// bank is completely replayed form blockstore, its fee stats can be made available to queries
+    fn finalize_block(&mut self, slot: Slot);
```

Comment:
Block finalization is a thing that already has a pretty specific meaning. Maybe we should use different words here?
Maybe just `finalize_prioritization_fee`